### DISCUSSION TITLE
Can't highlight date range in react-datepicker

### DISCUSSION THREAD

**Discussion by ramojej** [react-hook-form/react-hook-form#4106] [Q&A] [answered]
Hi, everyone! Got this problem that I'm unable to solve.. Also the startDate and endDate states are returning null..

Please see it [here](https://codesandbox.io/s/kind-kowalevski-ibjc7?file=/src/App.js).

**Comment by bluebill1049** [comment, marked as answer]
once you have integrated with `Controller`, you should use `setValue` instead of `useState`.

**Reply by ramojej** [reply]
I can't still make it work. Can you please explain it more? Should I setValue on onChange? Most of the docs are on click of a button. Also, where should I get the startDate and endDate prop in datepicker if I will not use useState?

**Reply by bluebill1049** [reply]
you are over complicated the Controller: https://codesandbox.io/s/fancy-pond-l30ob

**Reply by ramojej** [reply]
You're also not able to highlight the "range" of dates like in this one: [https://reactdatepicker.com/#example-date-range](https://reactdatepicker.com/#example-date-range)

**Reply by bluebill1049** [reply]
use `watch` and pass down those two selected value

**Reply by ramojej** [reply]
Got it working!! Thanks, @bluebill1049 !!!!!!

**Reply by bluebill1049** [reply]
awesome! no worries.
